lua/controller.lua

```diff
@@ -0,0 +1,412 @@
+		meta:set_string("jammed_item_name", src_name)
+		meta:set_float("times_ran_while_jammed", meta_times_ran_while_jammed + 1)
+	end
+	
+	return true
+end
+
+-- Set the controller definition using a table to allow for pipeworks and potentially other mod support
+local controller_def = {}
+controller_def.description = 'Drawer Controller'
+controller_def.tiles = {"drawer_controller_top_bottom.png", "drawer_controller_top_bottom.png", "drawer_controller_side.png", "drawer_controller_side.png", "drawer_controller_side.png", "drawer_controller_side.png"}
+controller_def.can_dig = controller_can_dig
+controller_def.groups = {cracky = 3, level = 2}
+controller_def.on_construct = function(pos)
+	local meta = core.get_meta(pos)
+	local inv = meta:get_inventory()
+	inv:set_size('src', 1)
+	meta:set_string("current_state", "running")
+	meta:set_float("times_ran_while_jammed", 0)
+	meta:set_string("jammed_item_name", "")
+	meta:set_string("drawers_table_index", "")
+	meta:set_string("formspec", controller_formspec(pos, "Running"))
+	local timer = core.get_node_timer(pos)
+	timer:start(7)
+end
+controller_def.on_blast = function(pos)
+	local drops = {}
+	default.get_inventory_drops(pos, "src", drops)
+	drops[#drops+1] = "drawers:controller"
+	core.remove_node(pos)
+	return drops
+end
+controller_def.on_timer = controller_node_timer
+controller_def.allow_metadata_inventory_put = controller_allow_metadata_inventory_put
+controller_def.allow_metadata_inventory_move = controller_allow_metadata_inventory_move
+controller_def.allow_metadata_inventory_take = controller_allow_metadata_inventory_take
+
+-- Mostly copied from the drawers in the drawer mod to add pipeworks support
+if core.get_modpath("pipeworks") and pipeworks then
+	controller_def.groups.tubedevice = 1
+	controller_def.groups.tubedevice_receiver = 1
+	controller_def.tube = controller_def.tube or {}
+	controller_def.tube.insert_object = function(pos, node, stack, tubedir)
+		local meta = core.get_meta(pos)
+		local inv = meta:get_inventory()
+		return inv:add_item("src", stack)
+	end
+	controller_def.tube.can_insert = function(pos, node, stack, tubedir)
+		local meta = core.get_meta(pos)
+		local inv = meta:get_inventory()
+		return inv:room_for_item("src", stack)
+	end
+	controller_def.tube.connect_sides = {left = 1, right = 1, back = 1, top = 1,
+		bottom = 1}
+	controller_def.after_place_node = pipeworks.after_place
+	controller_def.after_dig_node = pipeworks.after_dig
+end
+
+core.register_node('drawers:controller', controller_def)
+
+-- Because the rest of the drawers mod doesn't have a hard depend on default, I changed the recipe to have an alternative
+if core.get_modpath("default") and default then
+	core.register_craft({
+		output = 'drawers:controller',
+		recipe = {
+			{'default:steel_ingot', 'default:steel_ingot', 'default:steel_ingot'},
+			{'default:tin_ingot', 'group:drawer', 'default:copper_ingot'},
+			{'default:steel_ingot', 'default:steel_ingot', 'default:steel_ingot'},

```

Comment:
I think the controller should cost at least one mese crystal or diamond. And please also add a recipe for MCL2.